doc/sphinx/language/core/assumptions.rst

```diff
@@ -118,6 +120,10 @@ Function application
    | @term1
 
 :n:`@term1__fun @term1` denotes applying the function :n:`@term1__fun` to :token:`term1`.
+The type of :n:`@term1__fun` must be a product, (`forall ...`), which is the type of
```

Comment:
I like the formulation. Maybe recall here that `A -> B`  is equivalent to `forall x:A,B`?